File: src/backend_dotnet/HRMApi/Models/PointToMoneyHistory.cs
using System;
using System.Collections.Generic;

namespace HRMApi.Models;

public partial class PointToMoneyHistory
{
    public int Id { get; set; }

    public int EmployeeId { get; set; }

    public int PointRequested { get; set; }

    public decimal MoneyReceived { get; set; }

    public string? Status { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? ProcessedAt { get; set; }

    public virtual Employee Employee { get; set; } = null!;
}

File: src/backend_java/src/main/java/com/group07/human_resource_management/entity/PointToMoneyHistory.java
package com.group07.human_resource_management.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "point_to_money_history")
@Getter @Setter
@Builder @NoArgsConstructor @AllArgsConstructor
public class PointToMoneyHistory {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // FK employee_id
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "employee_id", nullable = false)
    private Employee employee;

    @Column(name = "point_requested", nullable = false)
    private Integer pointRequested;

    @Column(name = "money_received", nullable = false)
    private Double moneyReceived;

    private String status; // pending, approved, rejected, completed

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "processed_at")
    private LocalDateTime processedAt;
}
